matchmaker/views/matchmaker_api_tests.py

```diff
@@ -401,20 +408,27 @@ def test_search_individual_mme_matches(self, mock_nodes, mock_email, mock_slacke
     
     /project/R0001_1kg/family_page/F000001_1/matchmaker_exchange
     """
-        self.assertEqual(mock_slacker.call_count, 2)
+        self.assertEqual(mock_slacker.call_count, 3)
         mock_slacker.assert_called_with(MOCK_SLACK_TOKEN)
         slack_kwargs = {'as_user': False, 'icon_emoji': ':beaker:', 'username': 'Beaker (engineering-minion)'}
-        alert_slack_message = 'Error searching in Node A: Failed request (400)\n(Patient info: {})'.format(
-            json.dumps(expected_patient_body))
+        alert_a_slack_message = 'Error searching in Node A: Failed request (400)\n```{}```'.format(
+            json.dumps(expected_patient_body, indent=2))
+        alert_b_slack_message = 'Error searching in Node B: Received invalid results\n```{}```'.format(
+            json.dumps([INVALID_NEW_MATCH_JSON, INVALID_FEATURES_NEW_MATCH_JSON, INVALID_GENE_NEW_MATCH_JSON], indent=2))
         mock_slacker.return_value.chat.post_message.assert_has_calls([
-            mock.call('matchmaker_alerts', alert_slack_message, **slack_kwargs),
+            mock.call('matchmaker_alerts', alert_a_slack_message, **slack_kwargs),
+            mock.call('matchmaker_alerts', alert_b_slack_message, **slack_kwargs),
             mock.call('matchmaker_seqr_match', message, **slack_kwargs),
         ])
         mock_logger.error.assert_has_calls([
             mock.call(
                 'Slack error: Unable to connect to slack: Original message in channel ({}) - {}'.format(
-                    'matchmaker_alerts', alert_slack_message
+                    'matchmaker_alerts', alert_a_slack_message
             )),
```

Comment:
Mock and test the logger in the `communication_utils` module is a little weird. How about just mock `safe_post_to_slack`? And, maybe add some separate tests for the `communication_utils` module.

Reply:
no, this has better coverage of what actually happens and makes sure we adequately test when slack messages fail. I think removing test coverage is a terrible idea!